sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageConverter.cs

```diff
@@ -53,82 +52,80 @@ public static AmqpMessage BatchSBMessagesAsAmqpMessage(IReadOnlyCollection<Servi
-
             var batchMessages = new List<AmqpMessage>(source.Count);
             foreach (ServiceBusMessage sbMessage in source)
             {
-                if (firstAmqpMessage == null)
-                {
-                    firstAmqpMessage = SBMessageToAmqpMessage(sbMessage);
-                    firstMessage = sbMessage;
-                    batchMessages.Add(firstAmqpMessage);
-                }
-                else
-                {
-                    batchMessages.Add(SBMessageToAmqpMessage(sbMessage));
-                }
+                 batchMessages.Add(SBMessageToAmqpMessage(sbMessage));
             }
-
-            return BuildAmqpBatchFromMessages(batchMessages, firstMessage, forceBatch);
+            return BuildAmqpBatchFromMessages(batchMessages, forceBatch);
         }
 
         /// <summary>
         ///   Builds a batch <see cref="AmqpMessage" /> from a set of <see cref="AmqpMessage" />.
         /// </summary>
         ///
-        /// <param name="batchMessages">The set of messages to use as the body of the batch message.</param>
-        /// <param name="firstMessage">The first message being sent in the batch.</param>
+        /// <param name="source">The set of messages to use as the body of the batch message.</param>
         /// <param name="forceBatch">Set to true to force creating as a batch even when only one message.</param>
         ///
         /// <returns>The batch <see cref="AmqpMessage" /> containing the source messages.</returns>
         ///
-        private static AmqpMessage BuildAmqpBatchFromMessages(
-            List<AmqpMessage> batchMessages,
-            ServiceBusMessage firstMessage,
+        public virtual AmqpMessage BuildAmqpBatchFromMessages(
+            IReadOnlyCollection<AmqpMessage> source,
             bool forceBatch)
         {
             AmqpMessage batchEnvelope;
+            AmqpMessage firstMessage = null;
+
+            var batchMessages = source.GetEnumerator();
+            batchMessages.MoveNext();
 
-            if (batchMessages.Count == 1 && !forceBatch)
+            if (source.Count > 0)
             {
-                batchEnvelope = batchMessages[0];
+                firstMessage = batchMessages.Current;
+            }
+
+            if (source.Count == 1 && !forceBatch)
+            {
+                batchEnvelope = batchMessages.Current.Clone();
             }
             else
             {
-                var data = new List<Data>(batchMessages.Count);
-                foreach (var message in batchMessages)
+                var data = new List<Data>(source.Count);
+                var messageNum = 0;
+
+                foreach (var message in source)
                 {
                     message.Batchable = true;
                     using var messageStream = message.ToStream();
                     data.Add(new Data { Value = ReadStreamToArraySegment(messageStream) });
+
+                    ++messageNum;
```

Comment:
Why do we need this?

Reply:
Accidentally left over from a previous version.